tls/s2n_server_hello.c

```diff
@@ -270,6 +270,7 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
     POSIX_GUARD(s2n_server_hello_parse(conn));
 
     conn->actual_protocol_version_established = 1;
+    POSIX_GUARD_RESULT(s2n_conn_choose_state_machine(conn, conn->actual_protocol_version));
 
```

Comment:
s2n_conn_set_handshake_type is called by the server after processing the client hello and by the client after processing the server hello. I was worried that it would be too late in the case of the Hello Retry Request.

Reply:
Wouldn't the call in s2n_handshake_type_set_tls13_flag cover that case?

Reply:
Okay yes you are correct. There is another problem though, which is that conn_set_handshake_type branches based on the IS_TLS13_HANDSHAKE macro, which I have now changed to use conn->handshake.state_machine. So as it stands now, s2n_conn_choose_state_machine has to be called before s2n_conn_set_handshake_type.

Reply:
That sounds like a good argument for calling s2n_conn_choose_state_machine inside s2n_conn_set_handshake_type.